# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/PBackgroundSharedTypes.ipdlh
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 ipc/glue/PBackgroundSharedTypes.ipdlh
--- a/ipc/glue/PBackgroundSharedTypes.ipdlh	Tue May 21 22:51:54 2019 +0000
+++ b/ipc/glue/PBackgroundSharedTypes.ipdlh	Tue May 21 23:14:27 2019 +0000
@@ -32,8 +32,6 @@
 
   nsCString? domain;
 
-  ContentSecurityPolicy[] securityPolicies;
-
   // Like originNoSuffix, baseDomain is used out of the main-thread.
   nsCString baseDomain;
 };
@@ -61,5 +59,21 @@
   ExpandedPrincipalInfo;
 };
 
+struct PolicyInfo
+{
+  nsCString policy;
+  bool reportOnly;
+  bool deliveredViaMetaTag;
+};
+
+struct CSPInfo
+{
+  PolicyInfo[] policyInfos;
+  PrincipalInfo requestPrincipalInfo;
+  nsCString selfURISpec;
+  nsString referrer;
+  uint64_t innerWindowID;
+};
+
 } // namespace ipc
 } // namespace mozilla